core/file_walker.go

```diff
@@ -24,35 +29,53 @@ func (f *File) Path() string {
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
 func GetSubTree(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+	c := make(chan bool, maxConcurrentScans)
+	root := getSubTreeConcurrently(path, parent, readDir, ignoredFolders, c, &mutex, &wg)
+	wg.Wait()
+	return root
+}
+
+func getSubTreeConcurrently(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}, c chan bool, mutex *sync.Mutex, wg *sync.WaitGroup) *File {
 	ret := &File{}
 	entries, err := readDir(path)
 	if err != nil {
 		log.Println(err)
 		return ret
 	}
 	dirName, name := filepath.Split(path)
-	files := make([]*File, 0, len(entries))
-	var folderSize int64
+	ret.Files = make([]*File, 0, len(entries))
```

Comment:
Taking this back. I missed that you still allocated `len(entries)`. Actually you made it better.